frontend/src/views/PrinterCameraSetupPage.vue

```diff
@@ -0,0 +1,429 @@
+
+export default {
+  name: 'PrinterCameraSetupPage',
+
+  components: {
+    PageLayout,
+    StreamingBox,
+  },
+  data: function () {
+    return {
+      webcamStreamShutdown: false,
+      streamStarting: false,
+      printer: null,
+      webrtc: null,
+      webcams: [],
+      selectedWebcam: null,
+      selectedWebcamData: null,
+      streamMode: null,
+      h264HttpUrl: null,
+      rtspPort: null,
+      useRTSP: false,
+      isRaspi: false,
+      configuredCameras: [],
+    }
+  },
+
+  computed: {
+    isWebRTCCameraStreamer() {
+      if (!this.selectedWebcamData) return false
+      return this.selectedWebcamData.service === 'webrtc-camerastreamer'
+    },
+
+    streamConfig() {
+      if (!this.selectedWebcamData) return null
+
+      const config = { mode: this.streamMode }
+      if (this.streamMode === 'h264_copy') {
+        config.h264_http_url = this.h264HttpUrl
+      } else if (this.streamMode === 'h264_rtsp') {
+        config.rtsp_port = this.rtspPort
+      }
+
+      return config
+    },
+  },
+
+  watch: {
+    useRTSP: function (newValue, oldValue) {
+      this.webcamSelectionChanged()
+    },
+    isRaspi: function (newValue, oldValue) {
+      this.webcamSelectionChanged()
+    },
+  },
+
+  async created() {
+    const printerId = split(window.location.pathname, '/').slice(-3, -2).pop()
+    this.printer = await this.fetchPrinter(printerId)
+
+    this.getConfiguredWebcams()
+
+    this.printerComm = printerCommManager.getOrCreatePrinterComm(
+      printerId,
+      urls.printerWebSocket(printerId),
+      {
+        onPrinterUpdateReceived: null,
+        onStatusReceived: null,
+      }
+    )
+    this.printerComm.connect()
+  },
+
+  methods: {
+    async fetchPrinter(printerId) {
+      return axios.get(urls.printer(printerId)).then((response) => {
+        return normalizedPrinter(response.data)
+      })
+    },
+
+    fetchAgentWebcams() {
+      const webcamFetchOctoPayload = null // TODO
+      const webcamFetchMoonrakerPayload = {
+        func: 'server/webcams/list',
+        target: 'moonraker_api',
+      }
+
+      const webcamPayload = this.printer.isAgentMoonraker()
+        ? webcamFetchMoonrakerPayload
+        : webcamFetchOctoPayload
+
+      this.printerComm.passThruToPrinter(webcamPayload, (err, ret) => {
+        if (err) {
+          console.log(err, ret, '*****2')
+        } else {
+          this.webcams = ret?.webcams || []
+        }
+      })
+
+      const infoOctoPayload = null // TODO
+      const infoMoonrakerPayload = {
+        func: 'machine/system_info',
+        target: 'moonraker_api',
+      }
+      const infoPayload = this.printer.isAgentMoonraker() ? infoMoonrakerPayload : infoOctoPayload
+
+      this.printerComm.passThruToPrinter(infoPayload, (err, ret) => {
+        if (err) {
+          console.log(err, ret, '*****3')
+        } else {
+          this.isRaspi = ret.system_info.cpu_info.model.toLowerCase().includes('raspberry')
```

Comment:
How did you find out this is the right way to check if the agent is running on a raspberry pi? @aarontuftss

Reply:
From this documentation: https://moonraker.readthedocs.io/en/latest/web_api/#get-system-info